zerver/tests/test_subs.py

```diff
@@ -4413,6 +4437,35 @@ def test_user_settings_for_creating_web_public_streams(self) -> None:
             "create_web_public_stream_policy", invite_only=False, is_web_public=True
         )
 
+    def test_stream_creator_id(self) -> None:
+        iago = self.example_user("iago")
+        self.login_user(iago)
+        user1 = self.example_user("hamlet")
+        user2 = self.example_user("desdemona")
+
+        streams_to_sub = ["new_stream"]
+        # We create streams by subscribing users to non-existent streams
+        # Here we subscribe users other than the stream creator
+        with self.capture_send_event_calls(5) as events:
+            self.common_subscribe_to_streams(
+                iago,
+                streams_to_sub,
+                dict(principals=orjson.dumps([user1.id, user2.id]).decode()),
+            )
+        self.assertEqual(events[0]["event"]["streams"][0]["creator_id"], iago.id)
+        created_stream_id = events[0]["event"]["streams"][0]["stream_id"]
+
+        all_streams = self.api_get(iago, "/api/v1/streams")
+        json = self.assert_json_success(all_streams)
+        for stream in json["streams"]:
+            if stream["stream_id"] == created_stream_id:
+                # Acting user should be the creator for api created streams
+                self.assertEqual(stream["creator_id"], iago.id)
+                continue
+
+            # Streams that aren't created using the api should have no creator
+            self.assertIsNone(stream["creator_id"])
```

Comment:
It might be worth a follow-up change to `populate_db` to set creators for the initial streams, which might then require changing this test.

Reply:
Opened #30082 for this. Note that this doesn't fix the issue with `import_realm` tests not failing (mentioned below). 

#30081 added some tests and has some comments regarding that.